pywt/_extensions/_pywt.pyx

```diff
@@ -233,27 +241,41 @@ def families(int short=True):
     --------
     >>> import pywt
     >>> pywt.families()
-    ['haar', 'db', 'sym', 'coif', 'bior', 'rbio', 'dmey']
+    ['haar', 'db', 'sym', 'coif', 'bior', 'rbio', 'dmey', 'gaus', 'mexh', 'morl', 'cgau', 'shan', 'fbsp', 'cmor']
     >>> pywt.families(short=False)
-    ['Haar', 'Daubechies', 'Symlets', 'Coiflets', 'Biorthogonal', 'Reverse biorthogonal', 'Discrete Meyer (FIR Approximation)']
+    ['Haar', 'Daubechies', 'Symlets', 'Coiflets', 'Biorthogonal', 'Reverse biorthogonal', 'Discrete Meyer (FIR Approximation)', 'Gaussian', 'Mexican hat wavelet', 'Morlet wavelet', 'Complex Gaussian wavelets', 'Shannon wavelets', 'Frequency B-Spline wavelets', 'Complex Morlet wavelets']
 
     """
     if short:
         return __wfamily_list_short[:]
     return __wfamily_list_long[:]
 
 
-cdef public class Wavelet [type WaveletType, object WaveletObject]:
+def Wavelet(name=u"", object filter_bank=None):
```

Comment:
Changing `pywt.Wavelet` to a function instead of a class is useful because you can create both discrete wavelets and continuous wavelets from the name.

On the other hand, it's also an API break because `isinstance(w, pywt.Wavelet)` will no longer work. It might be best to have `Wavelet` be simple an alias to `DiscreteWavelet` (at least for now, perhaps deprecate it?), and then require explicit use of `ContinuousWavelet(name)`? I'm not sure...